packages/connectivity/src/scp-cf/destination/destination-service.ts

```diff
@@ -212,6 +212,62 @@ export async function fetchCertificate(
   }
 }
 
+/**
+ * Fetch a single destination by name without credentials.
+ * @internal
+ */
+export async function fetchDestinationByNameWithoutTokens(
+  destinationServiceUri: string,
+  serviceToken: string,
+  options: DestinationServiceOptions
+): Promise<Omit<Destination, 'AuthAndExchangeTokens'> | undefined> {
+  const targetUri = `${removeTrailingSlashes(
+    destinationServiceUri
+  )}/destination-configuration/v1/destinations/${
+    options.destinationName
+  }?$skipCredentials=true`;
+
+  const authHeader = wrapJwtInHeader(serviceToken).headers;
+
+  if (options?.useCache) {
+    const destinationsFromCache =
+      destinationServiceCache.retrieveDestinationsFromCache(
+        targetUri,
+        decodeJwt(serviceToken)
+      );
+    if (destinationsFromCache && destinationsFromCache.length > 0) {
+      logger.debug(
+        `Destinations retrieved from cache. There were ${destinationsFromCache.length} destinations returned from the cache.`
+      );
+      return destinationsFromCache[0];
+    }
+  }
+
+  return callDestinationEndpoint(targetUri, authHeader, options)
+    .then(response => {
+      const destination: Destination = parseDestination(response.data);
+      if (options.useCache) {
+        destinationServiceCache.cacheRetrievedDestinations(
+          targetUri,
+          decodeJwt(serviceToken),
+          [destination]
+        );
+      }
+      return destination;
+    })
+    .catch((error: AxiosError<any>) => {
+      if (error.response?.status === 404) {
```

Comment:
[q] For my understanding, could you please explain why we handle 404 differently and return undefined?

Reply:
My interpretation is that we use `undefined` as "The destination the user asked for does not exist", so we don't throw an error, but we also can't return a destination. The api will return 404 if the name of the destination does not exist.

Reply:
Comment added